### PR TITLE
Use HTML Parser to Colorize Text and add HTML Test

### LINKED ISSUE #1: Parse HTML with parser instead of RegExp
Currently a RegExp is used, which works for JSON and SQL, but not for languages where the tokens get nested like XML.

Could either use a basic HTML parser or construct a DOM and do a depth-first tree traversal to wrap the nodes.

### LINKED ISSUE #3: XML/HTML support
When trying to highlight an XML file, the `hljs` HTML tags are output.  Is this expected behaviour?  Using the `--language XML` option does not change the output.

Command:

`>highlight pom.xml`

Result:

```
<span class="hljs-tag"><project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd"></span>
        <span class="hljs-tag"><modelVersion></span>4.0.0<span class="hljs-tag"></modelVersion></span>
        <span class="hljs-tag"><parent></span>
                <span class="hljs-tag"><groupId></span>org.springframework.boot<span class="hljs-tag"></groupId></span>
                <span class="hljs-tag"><artifactId></span>spring-boot-starter-parent<span class="hljs-tag"></artifactId></span>
                <span class="hljs-tag"><version></span>1.5.2.RELEASE<span class="hljs-tag"></version></span>
        <span class="hljs-tag"></parent></span>
```

cli-highlight-3

### LINKED ISSUE #6: JSX support
Look like it doesn't work with JSX:

screen shot 2017-09-01 at 23 21 09

But you achieved a very good job!

### LINKED ISSUE #7: Highlighting ES6 file results in stray <span> elements
Cmd:
`$ highlight S3.js`

Output:
```node
// S3/index.js
// -----------
// Module for securely storing, retrieving, and deleting objects on S3.
import https        from 'https';
import S3SDK        from 'aws-sdk/clients/s3';
import { log }      from 'shared/log';
import APIInterface from 'shared/server/_lib/APIInterface';
import { uuid }     from 'shared/utilities';

export default <span class="hljs-class">class S3 extends APIInterface </span>{
  get requiredConfigKeys() {
    return ['bucket', 'endpoint', 'accessKeyId', 'secretAccessKey'];
  }

  constructor({ bucket, endpoint, accessKeyId, secretAccessKey } = {
    bucket: process.env.AWS_S3_BUCKET,
    endpoint: process.env.AWS_S3_ENDPOINT,
    accessKeyId: process.env.AWS_ACCESS_KEY_ID,
    secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
  }, createDuplicate = false) {
    const defaults = {
      region: 'us-east-1',
      sslEnabled: true,
      httpOptions: { agent: new https.Agent() },
      apiVersion: '2006-03-01',
    };
    super(Object.assign(defaults, { bucket, endpoint, accessKeyId, secretAccessKey }), false, createDuplicate);
  }
```

Perhaps this is a `highlight.js` issue?

### PR DESCRIPTION
**Pull request by keplersj**
Fixes #1 
Fixes #3 
Should Fix #6 
Might Fix #7

### CODE REVIEW

**Comment by keplersj** [comment]
So typings is super deprecated.. And I'm not sure how to reconcile that with this PR... Any recommendations @felixfbecker?

**Comment by felixfbecker** [comment]
Typings is deprecated now, so we should switch to `@types` packages

**Comment by keplersj** [comment]
How do you want to go about the switch? I can include it in this PR, do it in a separate PR, or wait for you to do it in master and rebase.

**Comment by felixfbecker** [comment]
I'm on it

File: src/index.ts [hunk 1]
```diff
@@ -1,15 +1,25 @@
 
 import * as hljs from 'highlight.js';
 import * as he from 'he';
```
Comment (felixfbecker) [inline src/index.ts]: Is this dependency still used after your change?

Comment (keplersj) [reply src/index.ts]: Nope.

**Comment by keplersj** [comment]
Updated and removed `he` dependency!

**Pull request merged**